crates/router/src/db/merchant_key_store.rs

```diff
@@ -90,6 +97,37 @@ impl MerchantKeyStoreInterface for Store {
             .change_context(errors::StorageError::DecryptionError)
         }
     }
+
+    async fn delete_merchant_key_store_by_merchant_id(
+        &self,
+        merchant_id: &str,
+    ) -> CustomResult<bool, errors::StorageError> {
+        let delete_func = || async {
+            let conn = connection::pg_connection_write(self).await?;
+            diesel_models::merchant_key_store::MerchantKeyStore::delete_by_merchant_id(
+                &conn,
+                merchant_id,
+            )
+            .await
+            .map_err(Into::into)
+            .into_report()
+        };
+
+        #[cfg(not(feature = "accounts_cache"))]
+        {
+            delete_func().await
+        }
+
+        #[cfg(feature = "accounts_cache")]
+        {
+            super::cache::publish_and_redact(
+                self,
+                CacheKind::Accounts(merchant_id.into()),
```

Comment:
This should be `format!("merchant_key_store_{}", merchant_id)`

Reply:
Changed to `let key_store_cache_key = format!("merchant_key_store_{}", merchant_id);`